File: leetCode Solutions/Q4_medianOfTwoSortedArrays/Q4_medianOfTwoSortedArrays.js
var findMedianSortedArrays = function (nums1, nums2) {

	nums1.sort();
	nums2.sort();

    let m = nums1.length;
    let n = nums2.length;
    
	let arr = nums1.concat(nums2);
	arr.sort(sortNumber);

	function sortNumber(a, b) {
		return a - b;
	}

    let x= (m+n - 1 )/2;
    let y= Math.floor(x);
	let z = Math.ceil(x);
	console.log("median of array: ", (m + n) / 2);
	console.log("array: ", arr);
    if(y == z){
		
		return (arr[x]);
    }
    else {
		return ((arr[y] + arr[z]) / 2);
    }
    
}; 
console.log(findMedianSortedArrays([-2, -1], [3]));

File: leetCode Solutions/Q4_medianOfTwoSortedArrays/Q4_medianOfTwoSortedArrays.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {

		int n1 = nums1.size();
		int n2 = nums2.size();

		vector<int> nums3(n1 + n2);

		merge(nums1.begin(), nums1.end(), nums2.begin(), nums2.end(), nums3.begin());

		sort(nums3.begin(), nums3.end());

		float x = ((float)n1 + (float)n2 - 1) / 2;
    	int y= floor(x);
		int z = ceil(x);

		if( y == z){
			return nums3[x];
		}
		else{
			return (((float)nums3[y] + (float)nums3[z]) / 2);
		}
	}
};